python/pyiceberg/types.py

```diff
@@ -422,7 +422,8 @@ class BooleanType(PrimitiveType):
         BooleanType()
     """
 
-    __root__ = "boolean"
+    def __init__(self) -> None:
+        super().__init__(__root__="boolean")
```

Comment:
Do we really need these constructors? I ran into issues with them in combination with Pydantic 2: https://github.com/apache/iceberg/pull/7782/files